src/Microsoft.AspNetCore.Mvc.Core/Internal/AuthorizationApplicationModelProvider.cs

```diff
@@ -33,33 +34,17 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            AuthorizationPolicy policy;
-
             foreach (var controllerModel in context.Result.Controllers)
             {
-                policy = AuthorizationPolicy.Combine(
-                    _authorizationOptions,
-                    controllerModel.Attributes.OfType<IAuthorizeData>());
-                if (policy != null)
-                {
-                    controllerModel.Filters.Add(new AuthorizeFilter(policy));
-                }
-
+                controllerModel.Filters.Add(new AuthorizeFilter(_policyProvider, controllerModel.Attributes.OfType<IAuthorizeData>()));
```

Comment:
Assuming we want the caching behavior, why can't we combine the policy here and create the AuthorizationFilter with the already built policy?